### ISSUE TITLE
Train and test set not from the same distribution?

### ISSUE THREAD

**Issue by kushalj001** [xliuhw/NLU-Evaluation-Data#7]
I am running into a very weird issue. When I use the train and test set as given in each fold, my intent classifier overfits on the data. However, if I split the trainset internally into a new train and test set, the model works fine. The same thing happens with the test set. Moreover, if I combine the train and test set, shuffle the examples, and create a new train and test split, my model still works fine.
This is making me think that the testing dataset for intent classification is out of distribution? Or there is a bug in my code.
Let me know if you have any insights about this.
Thank you!

**Comment by xliuhw** [comment]
@kushalj001 I don't know what the causes are in your application. The CV data were generated by the scripts, if I remember it correctly, it first generates a total number of utterances then split them into 10-folds (shuffled first), for each fold the trainset and testset are split by 80:20. Maybe there is something related to the sample size (in different intents) when generating the whole dataset? 

Please bear in mind that we didn't use those data to train and tune our own classifiers, I just use the trainset to train a model by the different platforms, then use the testset to test and evaluate the performances of the different platforms. I am not so sure whether it might make differences in our cases from your use cases in term of making use of the given generated CV datasets.

**Comment by kushalj001** [comment]
Yeah. I was making a very silly mistake during training. It is fine now!
Thank you!
